src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -145,6 +145,38 @@ export const migrations = {
       } catch (e) {
         throw new Error(`Failure attempting to migrate saved object '${doc.attributes.title}' - ${e}`);
       }
+    },
+    '7.1.0': (doc) => {
+      // [TSVB] Migrate percentile-rank aggregation (value -> values)
+      const visStateJSON = get(doc, 'attributes.visState');
+
+      if (visStateJSON) {
+        let visState;
+        try {
+          visState = JSON.parse(visStateJSON);
+        } catch (e) {
+          // Let it go, the data is invalid and we'll leave it as is
+        }
+        if (visState && visState.type === 'metrics') {
+          const series = get(visState, 'params.series') || [];
+
+          series.forEach(series => {
+            if (!isEmpty(series.metrics)) {
+              series.metrics.forEach(metric => {
+                if (metric.type === 'percentile_rank' && metric.value) {
+                  metric.values = [metric.value];
+
+                  delete metric.value;
+                }
+              });
+            }
+          });
+
+          doc.attributes.visState = JSON.stringify(visState);
```

Comment:
@chrisdavies Absolutely agree with you but in case of multiple modifications we cannot use return (see migration for  version 7.0.0 for example) 

I think the best solution will be something like: 

```

    '7.1.0': (doc) => {
      return [
        migrateSomething1, 
        migrateSomething2,
        migratePercentileRankAggregation
      ].reduce((newDoc, migrateFn) => migrateFn(newDoc), doc);
    }
```

For me it looks more friendly for developers and a little bit more maintainable

See updated PR

Reply:
The following API looks better: 

    '7.1.0': doc => executeMigrations([
        migrateSomething1, 
        migrateSomething2,
        migratePercentileRankAggregation
    ], doc)